### ISSUE TITLE
Create ``typevar-double-variance`` checker

### ISSUE THREAD

**Issue by DanielNoord** [PyCQA/pylint#5895]
### Current problem

The following code should raise a ``typevar-double-variance`` warning.

```python
T = TypeVar("T", covariant=True, contravariant=True) # [typevar-double-variance]
```

### Desired solution

Some test cases that can be used:

```python
T_co = TypeVar("T", covariant=True, contravariant=True) # [typevar-double-variance]
T_co = TypeVar("T", covariant=True, contravariant=False)
T_contra = TypeVar("T", covariant=True, contravariant=True) # [typevar-double-variance]
T_contra = TypeVar("T", covariant=False, contravariant=True)
T = TypeVar("T", covariant=False, contravariant=False)
```

**Comment by jpy-git** [comment]
@DanielNoord I'm going to add this as part of the work on #5224 since it's touching the same bit of code 😄

**Comment by DanielNoord** [comment]
If it isn't too much work for you I'd like you to ask you to separate the PRs. I know me and other maintainers have been quite busy the last couple of weeks and larger PRs just take more time to review. If split, it should still be pretty easy to rebase one of the PRs after the first has been merged.

That said, I'm very grateful for you taking on this issue 😄

**Comment by jpy-git** [comment]
Yeah that's no problem, in that case will do this one first since 😄
